core/src/main/java/com/datastrato/gravitino/utils/IsolatedClassLoader.java

```diff
@@ -214,11 +214,20 @@ private boolean isSharedClass(String name) {
         || name.startsWith("org.apache.log4j")
         || name.startsWith("org.apache.logging.log4j")
         || name.startsWith("java.")
-        || (name.startsWith("com.datastrato.gravitino.")
-            && !name.startsWith("com.datastrato.gravitino.catalog.hive."))
+        || !isCatalogClass(name)
         || sharedClasses.stream().anyMatch(name::startsWith);
   }
 
+  /**
+   * Checks if a given class name belongs to a catalog class.
+   *
+   * @param name The fully qualified class name.
+   * @return true if the class is a catalog class, false otherwise.
+   */
+  private boolean isCatalogClass(String name) {
```

Comment:
`isCatalogClass` means whether the provided name is a class name that is offered by specific catalogs like `Hive`, `Iceberg`, and so on.